File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        sum_digit = 0
        product_digit = 1
        num = n
        while num > 0:
            sum_digit += num % 10
            product_digit *= num % 10
            num //= 10
        return n % (sum_digit + product_digit) == 0

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int sumDigit = 0;
        int productDigit = 1;
        int num = n;
        while (num > 0) {
            sumDigit += num % 10;
            productDigit *= num % 10;
            num /= 10;
        }
        return n % (sumDigit + productDigit) == 0;
    }
}